portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,80 @@
 
 package com.google.sps.servlets;
 
+import com.google.sps.data.Comment;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
+import com.google.gson.Gson;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
+import java.text.SimpleDateFormat;  
+import java.util.Date; 
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+@WebServlet("/comments")
 public class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Ryan!");
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    String numComments = request.getParameter("num-comments");
+    int commentAmount = 0;
+    try{
+        commentAmount = Integer.parseInt(numComments);
+    }catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + numComments);
+      commentAmount = 5;
+    }
+
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable(FetchOptions.Builder.withLimit(commentAmount))) {
```

Comment:
Just wondering - is tracking the number of comments and fetching the exact num necessary or can you just fetch the whole list every time? It seems like the amount of comments in DataStore changes upon adds or deletes, so the count of comments in DataStore should equal your internal tracker for "num-comments'

Reply:
The num-comments refers to the number that user wants to display on the page  so basically this restricts the number of comments that I show to a smaller number. Should I retrieve all the comments from datastore here and then restrict the display in the JS?